src/algorithms/signal_source/libs/ion_gsms.h

```diff
@@ -0,0 +1,74 @@
+/*!
+ * \file ion_gsms.h
+ * \brief GNU Radio block that reads a Block from a file following ION's GNSS-SDR metadata standard
+ * \author Víctor Castillo Agüero, 2024. victorcastilloaguero(at)gmail.com
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * GNSS-SDR is a Global Navigation Satellite System software-defined receiver.
+ * This file is part of GNSS-SDR.
+ *
+ * Copyright (C) 2010-2020  (see AUTHORS file for a list of contributors)
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * -----------------------------------------------------------------------------
+ */
+
+#ifndef GNSS_SDR_ION_GNSS_SDR_METADATA_STANDARD_H
+#define GNSS_SDR_ION_GNSS_SDR_METADATA_STANDARD_H
+
+#include "configuration_interface.h"
+#include "gnss_block_interface.h"
+#include "ion_gsms_chunk_data.h"
+#include <gnuradio/block.h>
+#include <gnuradio/sync_block.h>
+#include <filesystem>
+#include <memory>
+#include <string>
+#include <vector>
+
+
+class IONGSMSFileSource : public gr::sync_block
+{
+public:
+    using sptr = gnss_shared_ptr<IONGSMSFileSource>;
+
+    IONGSMSFileSource(
+        const ConfigurationInterface* configuration,
```

Comment:
why do you define a signature with unused parameters? There should be nothing to be compatible with that requires a generic API here. In fact, as I understand the paradigm, the Adapter should translate from the Configuration to construct the blocks; the blocks should be completely insulated from the config mechanism.

Reply:
I didn't realize about that separation of configuration concerns between adapters and gnuradio_blocks. I figured it would be nice to have the configuration available in there just in case I needed it.

I will remove it.

Reply:
Fixed